### ISSUE TITLE
Create custom definition for Swagger only.

### ISSUE THREAD

**Issue by ShaneGMamet** [domaindrivendev/Swashbuckle#1044]
Hi

I was wondering if there is a way to create a custom definition name.

I would like to leave my live, running code as is, and by adding a Swagger attribute, have a model name within the "definitions" produced display a custom value.

i.e. 
A current definition of a model looks like this (mapped to a namespace):
	"definitions": {
		"This.Is.A.Long.Namespace.Model.Name": {}
I'd like to add an attribute that, when the Swagger is generated, it'll display:
	"definitions": {
		"CustomName": {}

See this question which explains more in detail: [http://stackoverflow.com/questions/42754857/customize-generated-model-names-swagger-ui](url)

**Comment by stefan-schweiger** [comment]
I also would really appreciate such a feature. Any chance we could get this with an class attribute?

**Comment by heldersepu** [comment]
Have you guys explored doing that with an IDocumentFilter?

**Comment by stefan-schweiger** [comment]
It's an valid option, but since most other things like Required, OperationId, etc. are set at their corresponding backend counterparts via attributes it think it would be more line with how things are currently "configured" to do this also with an attribute.

**Comment by heldersepu** [comment]
This got me thinking...
To me this is a low low priority feature and probably many contributors will agree.
But what if this is very important to others? 
- How can we measure importance ? 
- How do we come to an agreement ?

**Comment by heldersepu** [comment]
I think that https://opencollective.com/opensource tries to fix the problem I mentioned...
But I do not see any GitHub integration yet

**Comment by stefan-schweiger** [comment]
I agree that many users of this library don't need this or just live with the generated definition.

But because backend naming conventions (POCO, DAO, DTO, ViewModel, Entity suffix or SQL, Db, Data prefix) which are used by many companies should not necessarily be reflected to the outside api documentation in my opinion.

Doing this with a DocumentFilter requires at least one line of code for every Operation which references such a type and Hardcoding Statuscodes and Names which are a pain to refactor and not necessarily reflect the actual code.

**Comment by heldersepu** [comment]
I fully understand your concern, and 100% agree internal naming conventions should not be reflected to the outside.

**But how can we convince/motivate a developer to start working on this soon?**
My question is a bigger problem, I have seen it quite often with OpenSource projects.
I think that a monetary motivation can go a long way, what do you think?
